great_expectations/rule_based_profiler/types/builder.py

```diff
@@ -38,7 +39,16 @@ def __init__(
             batch_request: specified in DomainBuilder configuration to get Batch objects for domain computation.
         """
         self._batch_list = batch_list
+
+        if batch_request_contains_batch_data(batch_request=batch_request):
```

Comment:
I think we do want to be able to handle runtime batch requests in the builder so that we can enable the workflow of calling `get_domains()` in an interactive way as you are building a profiler.

Reply:
@anthonyburdi per our conversation, now there is a JIRA item for this.  Thanks. /cc @donaldheppner